yt/yt/server/job_proxy/environment.cpp

```diff
@@ -306,10 +306,10 @@ class TPortoUserJobEnvironment
 
         std::vector<TDevice> devices;
         for (const auto& descriptor : ListGpuDevices()) {
-            const auto& deviceName = descriptor.DeviceName;
-            if (std::find(Options_.GpuDevices.begin(), Options_.GpuDevices.end(), deviceName) == Options_.GpuDevices.end()) {
+            const auto& deviceIndex = descriptor.DeviceIndex;
+            if (std::find(Options_.GpuIndexes.begin(), Options_.GpuIndexes.end(), deviceIndex) == Options_.GpuIndexes.end()) {
                 devices.push_back(TDevice{
-                    .DeviceName = deviceName,
+                    .DeviceName = GetGpuDeviceName(deviceIndex),
```

Comment:
`DeviceName` is included in the descriptor next to the `DeviceIndex`, you could've just used it.

Reply:
Oh, probably we need some tests for this code that mock GPU device.